tvb_framework/tvb/adapters/uploaders/bids_importer.py

```diff
@@ -0,0 +1,279 @@
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2022, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+.. moduleauthor:: David Bacter <david.bacter@codemart.ro>
+.. moduleauthor:: Robert Vincze <robert.vincze@codemart.ro>
+"""
+
+import os
+import json
+import numpy
+
+from tvb.adapters.datatypes.db.region_mapping import RegionVolumeMappingIndex
+from tvb.adapters.datatypes.db.structural import StructuralMRIIndex
+from tvb.adapters.datatypes.db.time_series import TimeSeriesVolumeIndex
+from tvb.adapters.datatypes.db.volume import VolumeIndex
+from tvb.core.adapters.exceptions import LaunchException
+from tvb.core.entities.storage import dao
+from tvb.core.neotraits.forms import TraitUploadField
+from tvb.core.neotraits.uploader_view_model import UploaderViewModel
+from tvb.core.neotraits.view_model import Str
+from tvb.core.adapters.abcuploader import ABCUploaderForm, ABCUploader
+from tvb.datatypes.connectivity import Connectivity
+from tvb.datatypes.graph import CorrelationCoefficients
+from tvb.datatypes.surfaces import CorticalSurface
+from tvb.datatypes.time_series import TimeSeriesRegion
+
+
+class BIDSImporterModel(UploaderViewModel):
+    uploaded = Str(
+        label='BIDS derivatives dataset (zip)'
+    )
+
+
+class BIDSImporterForm(ABCUploaderForm):
+
+    def __init__(self):
+        super(BIDSImporterForm, self).__init__()
+
+        self.uploaded = TraitUploadField(BIDSImporterModel.uploaded, '.zip', 'uploaded')
+
+    @staticmethod
+    def get_view_model():
+        return BIDSImporterModel
+
+    @staticmethod
+    def get_upload_information():
+        return {
+            'uploaded': '.zip'
+        }
+
+
+class BIDSImporter(ABCUploader):
```

Comment:
Rename to BIDSDerivativesImporter?

Reply:
I consider this change, but I am not decided if it is a good idea or not.... 
We would then need to change also `BIDSImporterModel` and `BIDSImporterForm` into the more explicit names. Wouldn't these become too long ?

Reply:
On the other hand, TVB cannot read the official BIDS which describes empirical data...
Would it be enough to add a description near the upload field? (to explain that it expects derivatives)

Reply:
Is this similar to what you have in mind ?
![image](https://user-images.githubusercontent.com/674853/169819134-f16d24c0-4787-4db9-b344-b2bb4bee0485.png)

Reply:
Yes, this is informative enough from my POV